### PR TITLE
Allow custom image names for OSM Helm chart

### LINKED ISSUE #4331: Helm Chart should accommodate custom image names
**Please describe the Improvement and/or Feature Request**

The OSM Helm Chart should accommodate the use of custom image names.

**Scope (please mark with X where applicable)**
- New Functionality          [ ]
- Install                    [X ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- CI System                  [ ]
- Demo                       [ ]
- Project Release            [ ]

**Possible use cases**

The OSM Helm Chart should accommodate the use of custom image names. For example, some users may want to mirror the OSM images similar to how Rancher currently mirrors Istio images here: https://github.com/rancher/rancher/releases/download/v2.6.2/rancher-images.txt. A relevant extract from the rancher-images.txt file is included below. The current Helm Chart prevents the use of an image prefix like "mirrored-" in this example.
```
rancher/mirrored-istio-citadel:1.5.9
rancher/mirrored-istio-coredns-plugin:0.2-istio-1.1
rancher/mirrored-istio-galley:1.5.9
rancher/mirrored-istio-install-cni:1.10.4
rancher/mirrored-istio-kubectl:1.5.9
rancher/mirrored-istio-mixer:1.5.9
rancher/mirrored-istio-node-agent-k8s:1.5.9
rancher/mirrored-istio-pilot:1.10.4
rancher/mirrored-istio-pilot:1.5.9
rancher/mirrored-istio-proxyv2:1.10.4
rancher/mirrored-istio-proxyv2:1.5.9
rancher/mirrored-istio-sidecar_injector:1.5.9
```

### PR DESCRIPTION
**Pull request by shalier**
**Description**:
Map image name in values.yaml and resolves #4331

Signed-off-by: Shalier Xia <shalierxia@microsoft.com>

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| Install                    | [ x] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

### CODE REVIEW

File: charts/osm/values.schema.json [hunk 1]
```diff
@@ -244,12 +244,14 @@
                     "examples": [
                         {
                             "registry": "openservicemesh",
+                            "name": "osm-controller",
```
Comment (nojnhuh) [inline charts/osm/values.schema.json:247]: If `name` is supposed to map to an object then this doesn't seem valid. Given I'm not aware of anything parsing these examples, I'd say it's not worth adding any more to them here.

**Review by nojnhuh — APPROVED** [review]
Left one nit, otherwise looks good.

I verified that the output of `helm template` is the same before and after this change.

**Comment by jaellio** [comment]
Should we allow the registry and tag to also be set per image?

**Review by jaellio — APPROVED** [review]

**Comment by nojnhuh** [comment]
> Should we allow the registry and tag to also be set per image?

I think eventually it's likely we'll end up making this possible, though I think exactly how we should do that needs some extra thought to consider things like how backwards-compatible it should be with the way it is now.

**Pull request merged**